testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6538,6 +6586,54 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
         "INTEGER is not comparable to BOOLEAN", false);
   }
 
+  /** Test case for
+   * <a href="https://github.com/apache/calcite/pull/3705">[CALCITE-5976]
+   * Use explicit casting if inserted element type in ArrayPrepend/ArrayAppend/ArrayInsert
+   * does not equal derived component type)</a>. */
+  @Test void testArrayPrependFuncByCast() {
```

Comment:
This is all the tests I can think of, if anything is missing please let me know and I can make additional verifications